### PR TITLE
Add documentation and tests for insert_form.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Also adds some missing requires. More are still missing...

### CODE REVIEW

File: lib/forms/pegasus_form_validation.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@
 require_relative 'pegasus_form_errors'
 require_relative '../cdo/regexp'
 require_relative '../cdo/geocoder'
+require_relative '../../pegasus/forms/hoc_signup_2017'
```
Comment (aoby) [inline lib/forms/pegasus_form_validation.rb]: Why is this needed here? The `HocSignup2017` form doesn't appear to be used, and the tests associated with this module are passing.

Comment (ashercodeorg) [reply lib/forms/pegasus_form_validation.rb]: It was necessary for me to add, otherwise I see stacks like the following. Really, I think we need to be requiring all form kinds. In production, somehow these must already be visible from transitive requires?
```
  1) Error:
insert_form#test_0002_creates a forms_geos entry:
NameError: uninitialized constant HocSignup2017
    /Users/asher/code-dot-org/lib/forms/pegasus_form_validation.rb:133:in `const_get'
    /Users/asher/code-dot-org/lib/forms/pegasus_form_validation.rb:133:in `validate_form'
    /Users/asher/code-dot-org/pegasus/helpers/form_helpers.rb:34:in `insert_form'
    test/test_form_helpers.rb:65:in `block (2 levels) in <class:FormHelpersTest>'
```

File: pegasus/forms/hoc_signup_2014.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 require 'honeybadger'
+require_relative '../helpers/hourofcode_helpers'
```
Comment (ashercodeorg) [inline pegasus/forms/hoc_signup_2014.rb:2]: This defines `HOC_COUNTRIES` below.

File: lib/forms/pegasus_form_validation.rb (hunk 1 above)
Comment (aoby) [reply lib/forms/pegasus_form_validation.rb]: OIC. I think the form(s) should be required outside of `PegasusFormValidation`, in this case in your test. The reason it's complaining about the uninitialized constant `HocSignup2017` is because the test is [creating a form of that type](https://github.com/code-dot-org/code-dot-org/pull/16938/files#diff-046bf0d1021ea30f562b024992768939R69). 

Either the `test_form_helpers` test itself should require that form, or we should modify `PegasusFormHelpers` to take a class instead of kind (string), or otherwise provide a flexible way to supply the set of available forms, but I don't think it should require them directly. Part of my goal in extracting that code was to isolate it so it's more portable and testable.

Comment (aoby) [reply lib/forms/pegasus_form_validation.rb]: Admittedly it's a bit odd that it just attempts to find a class from the supplied name, and forms are meant to follow a pattern but don't even have a common base. I'd be happy if we improved that so it's more scoped.

Comment (ashercodeorg) [reply lib/forms/pegasus_form_validation.rb]: Moved the require from the validation file to the test file. Good call.

Comment (ashercodeorg) [reply lib/forms/pegasus_form_validation.rb]: Actually, I rescind the good call. The test is agnostic to the class, it is simply referencing the string `HocSignup2017`. Though I'm going to keep things as-is, I do think the validation is missing the require, and my tests are exposing this.

Comment (aoby) [reply lib/forms/pegasus_form_validation.rb]: In that case, I again think we should provide the validation a set of forms, and attempt to match from that. As it stands (leftover from the original design) it just tries to instantiate whatever class name you provide, which means the class name must exist in the context. The validation contract is that whatever name the caller provides should exist and have a `normalize` method.

Comment (aoby) [reply lib/forms/pegasus_form_validation.rb]: On that note, I'd be ok if the validator required the entire `forms` directory. I just don't think it should know or care about any specific form.
